modules/ocf_kubernetes/files/mastodon-pv.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: mastodon-nfs-pv
+spec:
+  capacity:
+    storage: 30Gi
+  accessModes:
+    - ReadWriteMany
+  nfs:
+    path: /opt/homes/services/mastodon
```

Comment:
What if we instead had `/opt/homes/services/kubernetes` and simply made mastodon a subdirectory there? This way, in the future, if we want to add more `nfs` stuff to kubernetes, we can simply create another directory and use one PV that we can r+w to.

Reply:
I think it's a better pattern to have separation between services, so we can modify them independently should it become necessary, and for general isolation as well.

Reply:
You can create this separation in the `PhysicalVolumeClaim`. See the mastodon one [here](https://github.com/fydai/mastodon/blob/master/kubernetes/mastodon-pv.yml). This includes access mode, amount of storage used, etc. The PV is just the backing for the storage mapping that  PVCs will request.

Reply:
What I said is wrong. There is a one-to-one mapping between PVs and PVCs, sadly. Instead, I would de-maston-ify this and we can put all future PVs in this file, separated by `--`. This will reduce the `kubectl` code in puppet.